### DISCUSSION TITLE
JSON not formatted anymore

### DISCUSSION THREAD

**Discussion by bytetec** [FLAMEGPU/FLAMEGPU2#1204] [Q&A] [answered]
Hi everyone
Since the update from FLAME GPU 2.0.0-rc to FLAME GPU 2.0.0-rc.1 the export of the logfile in JSON produces only a one liner. Does anybody else has this issue?
Before:
![image](https://github.com/FLAMEGPU/FLAMEGPU2/assets/138150718/6118a5d0-d5d7-4746-921e-c0d4dc54f466)
After:
![image](https://github.com/FLAMEGPU/FLAMEGPU2/assets/138150718/aa7ce8ba-f84d-40dd-8720-4edf7d40d745)

**Comment by mondus** [comment]
Hi @bytetec. The json logger has not been updated for quite a while but we have bumped up the version of rapidjson in our build. We will need to investigate this further as there is no obvious bug reported upstream in rapidjson.

**Comment by Robadob** [comment, marked as answer]
Hi,

This does appear to be a bug. As noted in #1205, when `JSONStateWriter` was overhauled to allow partial agent state to be exported the handling of RapidJSON's `Writer`/`PrettyWriter` was changed from using templating to C++ inheritance.

However it appears, although `PrettyWriter` is a subclass of `Writer` it's methods are none-virtual. What this means is that if we cast a `PrettyWriter*` to `Writer*` and use the writing methods, they are not forwarded to the `PrettyWriter` subclass.

Thanks for spotting it, our test suite doesn't currently test the status of output files for the pretty print setting being observed.

*It's low priority and a bit of a grim fix. I've not formally got any time allocated for maintaining FLAMEGPU currently, so I can't promise when it will be resolved.*
